services/sequelize/recovery_method.js

```diff
@@ -0,0 +1,169 @@
+const { Op } = require('sequelize');
+
+const models = require('../../models');
+
+const { Account } = models;
+
+const SequelizeRecoveryMethods = {
+    cleanRecoveryMethod(recoveryMethod) {
+        if (!recoveryMethod) {
+            return null;
+        }
+
+        const {
+            kind,
+            detail,
+            publicKey,
+            requestId,
+            securityCode,
+            createdAt,
+            updatedAt,
+        } = recoveryMethod.toJSON();
+
+        return {
+            kind,
+            detail,
+            publicKey,
+            requestId: parseInt(requestId, 10),
+            securityCode,
+            createdAt,
+            updatedAt,
+        };
+    },
+
+    async createRecoveryMethod({ accountId, detail, kind, publicKey, requestId, securityCode }) {
+        const account = await Account.findOne({ where: { accountId } });
+        const recoveryMethod = await account.createRecoveryMethod({
+            kind,
+            ...(detail && { detail }),
+            ...(publicKey && { publicKey }),
+            ...(requestId && { requestId }),
+            ...(securityCode && { securityCode }),
+        });
+
+        return this.cleanRecoveryMethod(recoveryMethod);
+    },
+
+    async deleteOtherRecoveryMethods({ accountId, detail }) {
+        const account = await Account.findOne({ where: { accountId } });
+        if (!account) {
+            return null;
+        }
+
+        const allRecoveryMethods = await account.getRecoveryMethods();
+        for (const rm of allRecoveryMethods) {
+            if (rm.detail !== detail) {
+                await rm.destroy();
+            }
+        }
+    },
+
+    async deleteRecoveryMethod({ accountId, kind, publicKey }) {
+        const account = await Account.findOne({ where: { accountId } });
+        const [recoveryMethod] = await account.getRecoveryMethods({
+            where: {
+                kind,
+                publicKey,
+            }
+        });
+
+        return recoveryMethod.destroy();
+    },
+
+    async getTwoFactorRecoveryMethod(accountId) {
+        const twoFactorRecoveryMethod = await this.getTwoFactorRecoveryMethod_internal(accountId);
+        return this.cleanRecoveryMethod(twoFactorRecoveryMethod);
+    },
+
+    async getTwoFactorRecoveryMethod_internal(accountId) {
```

Comment:
Just thinking out loud -- node >= 14 supports private class methods and Render uses node v14.17...if we continue using it we could just mark this as private.  Not a blocker, though, we can come back to it.

Reply:
You'd still have collisions with the public method, no? Anyway in this context I used `_internal` to indicate methods that return ORM objects as opposed to POJOs - I only wanted to return POJOs in the service objects so I moved all that jank under `/sequelize` 😎

Reply:
True on the method name conflict, 🤦  :)
Roger that on the _internal naming due to the rich object vs. POJO :+1: I can dig it.